tamr_unify_client/dataset/collection.py

```diff
@@ -171,6 +171,189 @@ def create_from_dataframe(
 
         return dataset
 
+    def create_from_geodataframe(
+        self, geodf, primary_key_name, dataset_name, geo_attr="geometry"
+    ):
+        """Creates a dataset in this collection with the given name, creates an attribute for each column in the `geodf`
+        (with `primary_key_name` as the key attribute), and upserts a record for each row of `geodf`.
+
+        Each attribute has the default type `ARRAY[STRING]`, besides the key attribute, which will have type `STRING`,
+        and the geometry attribute `geo_attr` which will have the type `GEOSPATIAL`.
+
+        This function attempts to ensure atomicity, but it is not guaranteed. If an error occurs while creating
+        attributes or records, an attempt will be made to delete the dataset that was created. However, if this
+        request errors, it will not try again.
+
+        :param geodf: The data to create the dataset with.
+        :type df: :class:`geopandas.geodataframe.GeoDataFrame`
```

Comment:
Type information should be encoded as type annotations (see #300 , for an example see https://github.com/Datatamer/tamr-client/blob/master/tamr_unify_client/dataset/resource.py#L89 )

Reply:
In order to specify the type `geodf: gpd.GeoDataFrame` I would need to import geopandas as in the [example](https://github.com/Datatamer/tamr-client/blob/6924c26975e66e58933ee75acde48fa68e2bc09a/tamr_unify_client/dataset/resource.py#L3)

Are we happy to do that? It would create a dependency on geopandas.

Reply:
Causes checks to fail since the package is not  installed in pytest.

Reply:
I think a dependency on GeoPandas is ok here as long as there aren't any gotchas for installation. @tiems90 is GeoPandas easy to install? (just a `pip install` or does it require other stuff?)

In the `tamr_client` BETA, we have been packaging things by *feature* and have been planning to include optional features (see #351 ) so we could include GeoPandas as an optional feature there if needed. But I think its worth straight up including as a dependency first. If people have issues with GeoPandas as a dependency, they can file issues on this repo and then we can make into an optional dependency.

Reply:
Ok! A pip install should do.
Some of the functionality requires additional libraries installed `apt-get install` style, though.

Reply:
Added `geopandas` to poetry. It now passes the checks.